core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java

```diff
@@ -28,6 +28,47 @@
  * its name.
  *
  * <p>The server is intended to be fully-featured, high performance, and useful in testing.
+ *
+ * <h3>Using JUnit TestRule</h3>
+ * The class "GrpcServerRule" (from the "grpc-java/testing") is a JUnit TestRule that
+ * creates a {@link InProcessServer} and a {@link io.grpc.ManagedChannel ManagedChannel}. This
+ * test rule contains the boiler plate code shown below. The classes "HelloWorldServerTest" and
+ * "HelloWorldClientTest" (from "grpc-java/examples") demonstrate basic usage.
+ *
+ * <h3>Usage example</h3>
+ * <h4>Server and client channel setup</h4>
+ * <pre>
+ *   MutableHandlerRegistry serviceRegistry = new MutableHandlerRegistry();
+ *   addServicesHelper(serviceRegistry); // your code here
+ *   ExecutorService serverExecutor = Executors.newCachedThreadPool();
+ *   Server server = InProcessServerBuilder.forName("unique-name")
+ *       .executor(serverExecutor)
+ *       .fallbackHandlerRegistry(serviceRegistry)
+ *       .build().start();
+ *   ExecutorService clientExecutor = Executors.newCachedThreadPool();
```

Comment:
It's a test, not a real production service. It is a Good Idea™ to use directExecutor() in tests when possible; it's not generally good to do that in production.

And people probably aren't going to shutdown the executors, so it is sort of more harmful to specify them in addition to being additional, unnecessary boilerplate.